contracts/YieldCrabLever.sol

```diff
@@ -0,0 +1,451 @@
+
+    /**
+     * @notice get the vault composition of the strategy
+     * @return operator
+     * @return nft collateral id
+     * @return collateral amount
+     * @return short amount
+     */
+    function getVaultDetails()
+        external
+        view
+        returns (
+            address,
+            uint256,
+            uint256,
+            uint256
+        );
+
+    /**
+     * @notice flash deposit into strategy, providing ETH, selling wSqueeth and receiving strategy tokens
+     * @dev this function will execute a flash swap where it receives ETH, deposits and mints using flash swap proceeds and msg.value, and then repays the flash swap with wSqueeth
+     * @dev _ethToDeposit must be less than msg.value plus the proceeds from the flash swap
+     * @dev the difference between _ethToDeposit and msg.value provides the minimum that a user can receive for their sold wSqueeth
+     * @param _ethToDeposit total ETH that will be deposited in to the strategy which is a combination of msg.value and flash swap proceeds
+     * @param _poolFee Uniswap pool fee
+     */
+    function flashDeposit(uint256 _ethToDeposit, uint24 _poolFee)
+        external
+        payable;
+
+    /**
+     * @notice flash withdraw from strategy, providing strategy tokens, buying wSqueeth, burning and receiving ETH
+     * @dev this function will execute a flash swap where it receives wSqueeth, burns, withdraws ETH and then repays the flash swap with ETH
+     * @param _crabAmount strategy token amount to burn
+     * @param _maxEthToPay maximum ETH to pay to buy back the wSqueeth debt
+     * @param _poolFee Uniswap pool fee
+
+     */
+    function flashWithdraw(
+        uint256 _crabAmount,
+        uint256 _maxEthToPay,
+        uint24 _poolFee
+    ) external;
+}
+
+/// @title A contract to help users build levered position on crab strategy token
+///        using ETH/WETH/DAI/USDC as collateral
+/// @author iamsahu
+/// @notice Each external function has the details on how this works
+contract YieldCrabLever is YieldLeverBase {
+    using TransferHelper for IERC20;
+    using TransferHelper for IMaturingToken;
+    using CastU128I128 for uint128;
+    using CastU256U128 for uint256;
+
+    ICrabStrategy public constant crabStrategy =
+        ICrabStrategy(0x3B960E47784150F5a63777201ee2B15253D713e8);
+
+    ISwapRouter public constant swapRouter =
+        ISwapRouter(0xE592427A0AEce92De3Edee1F18E0157C05861564);
+    IERC20 public crab;
+    bytes6 public constant crabId = 0x333800000000;
+    bytes6 public constant wethId = 0x303000000000;
+    bytes6 public constant daiId = 0x303100000000;
+    bytes6 public constant usdcId = 0x303200000000;
+
+    constructor(Giver giver_) YieldLeverBase(giver_) {}
+
+    /// @notice Invest by creating a levered vault.
+    /// The steps are as follows:
+    /// 1. Based on the ilk transfer it from the user
+    /// @param seriesId The series to invest in
+    /// @param ilkId The collateral to use
```

Comment:
I'm confused, the vault that you are building is crab-collateralized. What is the `ilk` here?